src/components/Form/index.js

```diff
@@ -0,0 +1,184 @@
+import React, { Component } from 'react'
+import dot from 'dot-object'
+
+export default class Form extends Component {
+   static propTypes = {
+      children: React.PropTypes.node,
+      onError: React.PropTypes.func,
+      onSubmit: React.PropTypes.func
+   }
+
+   constructor(props) {
+      super(props)
+
+      this.handleSubmit = this.handleSubmit.bind(this)
+
+      this.initialValuesLoaded = false
+
+      // Setup state
+      this.state = {}
+      props.children.forEach((child) => {
```

Comment:
Yeah I sort of agree. I went down that path for a short bit initially. Let's chat about this today?

Also, `isPristine` may not have to be part of the 1st level form state, it's an attribute owned by the children elements.